src/main/docs/guide/config/immutableConfig.adoc

```diff
@@ -55,6 +55,26 @@ snippet::io.micronaut.docs.config.immutable.VehicleSpec[tags="start",indent=0,ti
 
 The above example prints: `"Ford Engine Starting V8 [rodLength=7B.0]"`
 
+==== Using Java Record Classes to define immutable configuration
+
+For Java language applications, it's also possible to use https://docs.oracle.com/en/java/javase/17/language/records.html[Java Record Classes] for immutable configuration with ann:context.annotation.ConfigurationProperties[]. For example:
+
+[source, java]
+----
+import io.micronaut.context.annotation.ConfigurationProperties;
```

Comment:
this should be in code not in asciidoctor and we should have a test for the code shown in the documentation.